### ISSUE TITLE
Bug in `curry`

### ISSUE THREAD

**Issue by lambdatastic** [cullophid/barely-functional#32]
Hi! I was poking around around a bit and noticed that the way `curry` is written right now doesn't seem to work as intended

```javascript
const add4 = curry((a,b,c,d) => a+b+c+d)

const test = add4(1) // returns a function

test(2) // returns a function
test(3) // returns a function
test(4) // incorrectly returns 10
```

if you stick logs in the function as such:

```javascript
const curry = (f) => function () {
  let args = []
  const curried = function () {
    console.log(args)
    args = args.concat(Array.prototype.slice.call(arguments))
    console.log(args)
    return (args.length >= f.length) ? f.apply(null, args) : curried
  }
  return curried.apply(null, arguments)
}
```

you'll see that the intermediate calls are mutating args instead of returning a new closure:

```
>> test = add4(1)
[]
[ 1 ]
=> [Function]
>> test(2)
[ 1 ]
[ 1, 2 ]
=> [Function]
>> test(3)
[ 1, 2 ]
[ 1, 2, 3 ]
=> [Function]
>> test(4)
[ 1, 2, 3 ]
[ 1, 2, 3, 4 ]
=> 10
```

**Comment by chrfrasco** [comment]
Fixed in #33

**Comment by cullophid** [comment]
merged
